Support Thermo3 and Expand2 Clickboards

cdomnik opened the pull request:
Information
------------------------------------------------------------
### 1. Added Task for Thermo3Click and Expand2Click.
### 2. Adapt Weblayout to view the generated data.
### _**NOTE:**_ Color2Click will not work with this Branch by default. You will have to change the configuration to use your Color2Click board
------------------------------------------------------------

Bilder
--------------------------------------------------------------------------------------------------------------------
![screenshot_thermo3](https://user-images.githubusercontent.com/30650575/29125012-d6222a66-7d1a-11e7-9931-a27515503619.png)
--------------------------------------------------------------------------------------------------------------------
![screenshot_expand2](https://user-images.githubusercontent.com/30650575/29125015-dd375efc-7d1a-11e7-959e-09beef4df46a.png)

cdomnik committed:
Added Termo3 and Expand2 features

Changes:
 - Added Thermo3Click Board to front and back end
   - Temerature is shown with min, max, current and history

 - Added Expand2Click Board to frond and back end
   - water meter function

cdomnik committed:
Merge pull request #1 from cdomnik/clickboard features added

Added Termo3 and Expand2 features

cdomnik committed:
Changed some project settings

mlamberti commented on Applications/Clickboards/src/Expand2Click.c:
```diff
@@ -84,55 +84,148 @@ void set_expand2click(char data){
+		}
+
+		//oBits++;
+		//toggle obits each time the Task is called
```
Why don't toggle both bits in one go, without 'swap'?
oBits ^= 0x03;

mlamberti commented on Applications/Clickboards/src/Expand2Click.c:
```diff
@@ -84,55 +84,148 @@ void set_expand2click(char data){
+						 240,
+						 NULL,
+						 ( tskIDLE_PRIORITY + 1 ),
+						 xClickTaskHandleExpand )
```
xTaskCreate requires a pointer to a TaskHandle_t, thus line 187:
`&xClickTaskHandleExpand )`

mlamberti commented on Applications/Clickboards/src/thermo3click.c:
```diff
@@ -3,82 +3,172 @@
+						 300,
+						 NULL,
+						 ( tskIDLE_PRIORITY + 1 ),
+						 xClickTaskHandleThermo )
```
Same as above, should be a pointer: `&xClickTaskHandleThermo`

mlamberti commented on Applications/HTTP/httpd-fs/main.js:
```diff
@@ -83,18 +83,79 @@ templates['color2'] = `
+        			<td>{{temp}}&degC</td>
+        		</tr>
+        		<tr>
+        			<td>H&oumlchste Temperatur</td>
```
I'm not sure wether it's necessary to escape HTML entities, but it must be `&ouml;`

mlamberti replied:
Also `&deg;C`

mlamberti commented on Applications/HTTP/httpd-fs/main.js:
```diff
@@ -83,18 +83,79 @@ templates['color2'] = `
+        		</table>
+        	</p>
+        	<h3>Output Register</h3>
+        	not jet implemented
```
yet

mlamberti commented on Applications/HTTP/httpd-fs/main.js:
```diff
@@ -83,18 +83,79 @@ templates['color2'] = `
+        				<td>{{bits}}</td>
+        			</tr>
+        			<tr>
+        				<td>Gez&aumlhlte Wassermenge seit Reset</td>
```
`&auml;`

mlamberti commented on Applications/HTTP/httpd-fs/main.js:
```diff
@@ -83,18 +83,79 @@ templates['color2'] = `
 
 function toggleLED() {
-  	xhr = $.getJSON('status.json?action=set&led=' + ($('#led-switch').is(":checked") ? 'on' : 'off'), function(json) {});
+  	xhr = $.getJSON('http://192.168.1.118/' + 'status.json?action=set&led=' + ($('#led-switch').is(":checked") ? 'on' : 'off'), function(json) {});
```
The URI used for testing purposes shouldn't be part of the pull request.

mlamberti commented on Applications/HTTP/httpd-fs/main.js:
```diff
@@ -138,6 +205,48 @@ function processJSON(page, json) {
+			var checked3 = "";
+			var checked2 = "";
+			var checked1 = "";
+			var checked0 = "";
```
Those variables (checked0-7) are unused.

mlamberti commented on Applications/HTTP/httpd-fs/main.js:
```diff
@@ -138,6 +205,48 @@ function processJSON(page, json) {
+			if (( val / 8 ) >= 1 ) { json['checked3'] = "checked"; val = val % 8; }
+			if (( val / 4 ) >= 1 ) { json['checked2'] = "checked"; val = val % 4; }
+			if (( val / 2 ) >= 1 ) { json['checked1'] = "checked"; val = val % 2; }
+			if ( val == 1 ) { json['checked0'] = "checked"; }
```
Make it a loop:
```
for( var i = 0; i < 8; i++; ) {
    if( json['bits'] & (1<<i) ) json['checked'+i] = "checked";
}
```

mlamberti reviewed (changes requested):
Hey Christoph,
ich habe deine Änderungen durchgesehen und noch ein paar Bugs gefunden bzw. Bemerkungen geschrieben. Kannst du die Sachen bitte noch ändern?
Danke, Michael

cdomnik committed:
Minor changes and Bugfixes

Description:
 - thermo3Click.c
   > set timerdelay to 10 seconds
   > use define to configure delay
   > changed taskCreate(), task handle is now given as pointer
 - thermo3Click.h
   > added define to configure delay

 - Expand2Click.c
   > set timer delay to 100ms
   > use define to configure delay
   > use XOR to toggle Bits instead of if structure
   > changed taskCreate(), task handle is now given as pointer
 -Expand2Click.h
   >added define to configure delay

 - main.js
   > Fixed Bugs and Typos
   > Delay of Thermo3 calling switchPage() is now 10 seconds
   > Delay of Expand2 calling switchPage() is now 500ms
   > Expand2 Clickboard page is now rendered with a for loop
   > Removed IP in $.get, was just used for Debuging

NOTE: color2click will not work with this build, if you want to use it,
please change ClickboardConfig.c and clickboard_config.c!

cdomnik committed:
Merge pull request #2 from cdomnik/temp_stable

Update develop branch with bug fixes and minor changes

cdomnik commented:
Hi Michael,
habe die gewünschten Änderungen umgesetzt. Außerdem habe ich auf Wunsch von Thomas die Verzögerung des Expand Tasks auf 100ms runter gesetzt. Sollte passen mit dem Timing, hatte nur manchmal Timingprobleme, wenn ich die Webseite alle 100ms neu aufbaue.
Gruß, Christoph
